src/utils/io/progress_log.rs

```diff
@@ -0,0 +1,235 @@
+// Copyright 2019-2023 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+use humantime::format_duration;
+use std::time::{Duration, Instant};
+
+use parking_lot::Mutex;
+use std::io;
+use std::pin::Pin;
+use std::sync::Arc;
+use std::task::{Context, Poll};
+use tokio::io::ReadBuf;
+
+use log::info;
+
+const UPDATE_FREQUENCY: Duration = Duration::from_millis(5000);
+
+#[allow(dead_code)]
+pub fn wrap_iter<Inner>(
+    message: &str,
+    into_iter: impl IntoIterator<IntoIter = Inner>,
+) -> WithProgressIter<Inner>
+where
+    Inner: Iterator,
+{
+    let inner = into_iter.into_iter();
+    let (lower_bound, upper_bound) = inner.size_hint();
```

Comment:
I'm guessing we want to get rid of this as well in favour of something explicit, like `with_total_size`. Unless I'm missing something - size hint is not a precise indication, so the progress tracking is going to be very confusing.

Reply:
This code was removed since it was sitting unused.